### ISSUE TITLE
Bring back old Centered Layout behavior

### ISSUE THREAD

**Issue by jechase** [Microsoft/vscode#53989]
When first introduced, Centered Layout would only be active if there was a single editor (or if there was a horizontal split maybe? I never used that layout, so idk). If, while in the Centered Layout, a "split vertical" command was issued, the centered layout would be disabled until there was only one editor again.

In the most recent release (1.25.0), this behavior was changed such that, rather than disabling the centered layout while there's a vertical split, the editor stays centered with the same dimensions, and the split occurs inside this centered container. There was something in the release notes about how the old behavior was confusing or had unexpected results, but no details on what the actual unexpected behavior was.

I'm having trouble thinking of a situation in which this is ever what you want. For the Centered Layout to be most effective, the width of the editor should be close to the width of the actual text being edited, that way the text is actually centered in your view. When I say "give me another editor next to my current one," I *never* want two editors that are half the width of the text that I'm editing, but that's how the new centered layout behaves. You'd have to either
1. Resize the centered container for the new split layout and then resize it again back to the original dimensions once you're done with the split
2. Manually disable Centered Layout after splitting and manually re-enable it when you're back to one editor or
3. Keep the width of your centered window large enough to accommodate a split. But then it's not actually centered.

If not a full reversion back to the old behavior, can we at least get a configuration option to re-enable it? [Obligatory XKCD](https://xkcd.com/1172/)

Centered:
![2018-07-10-155920_1914x1032_scrot](https://user-images.githubusercontent.com/1553581/42534262-467ca57e-845a-11e8-8e09-e801da959078.png)

Split:
![2018-07-10-155935_1914x1032_scrot](https://user-images.githubusercontent.com/1553581/42534272-4dd513e2-845a-11e8-8e46-5922b9aa21be.png)

**Comment by isidorn** [comment]
@jechase thanks for the xkcd, that is one of my favorites.
I would only consider introducing an option if more users ask for this. So for now leaving this open and waiting for feedback.

fyi @joaomoreno

**Comment by Ky6uk** [comment]
Related to #53694. I am using only single layout with always enabled and centered zen-mode for a long time. `v1.25` just adds some [extra gaps](https://github.com/Microsoft/vscode/issues/53694#issue-338896113) around tabs and nothing more. Also (according to #53694) the option `zenMode.centerLayout` [has no effect at all](https://github.com/Microsoft/vscode/issues/53694#issuecomment-405276250), no matter `false` it or `true`.

I know Zen-mode isn't in priority for some reasons, but I hope that mode will be changed in near future. And Zen-mode will be really Zen-able, [flexible](https://github.com/Microsoft/vscode/issues/45617) and [clean](https://github.com/Microsoft/vscode/issues/48811) (like in Atom editor, for example).

**Comment by isidorn** [comment]
Yes, we will consider bringing back some of the old behavior in the future.
fyi @joaomoreno

**Comment by uloco** [comment]
IMHO the centered layout option should remember its position respecting open vertical split editors
* A single window
* Two splits
* Tree splits
etc...

So I would not have to resize my centered layout width every time I change the amount of opened splits.

**Comment by Ky6uk** [comment]
@isidorn will it fix the ~regression~ as-designed feature (#53694) when tabs bar has a reduced size like an editor pane? 🤔

**Comment by bpasero** [comment]
@isidorn should this rather be `workbench.editor.centeredLayoutAutoResize`?

**Comment by isidorn** [comment]
@Ky6uk no, it will not. Sorry
@bpasero yeah, editor prefix makes change looking at other settings whcich have the editor prefix. Updating it...
